src/multivariate/product.jl

```diff
@@ -20,6 +21,10 @@ struct Product{
         V<:AbstractVector{T} where
         T<:UnivariateDistribution{S} where
         S<:ValueSupport
+        Base.depwarn(
+            "`Product(v)` is deprecated, please use `product_distribution(v)`",
+            :Product,
+        )
```

Comment:
Maybe I'm just brainfarting, but it seems like this combined with the `@deprecate` below means that no matter which constructor I use for a vector of univariate distributions, I'm going to get a deprecation-warning?

Reply:
Yes, at the moment one always get's a depwarn ... #1589

Reply:
Yeah it's unfortunate and fixed by https://github.com/JuliaStats/Distributions.jl/pull/1590. I thought someone would approve the PR and I would be able to make a bugfix release shortly after the issue was discovered but it seems nobody has approved it within almost two weeks. I'm going to merge it and tag a release now since the issue is quite annoying for downstream packages (and AFAICT even causes time outs in Turing).

Reply:
Thanks @devmotion !

Reply:
Wonderful stuff @devmotion !